lib/vtls/schannel.c

```diff
@@ -672,12 +691,13 @@ schannel_connect_step2(struct connectdata *conn, int sockindex)
     infof(data, "schannel: SSL/TLS handshake complete\n");
   }
 
-#ifdef _WIN32_WCE
-  /* Windows CE doesn't do any server certificate validation.
-     We have to do it manually. */
-  if(conn->ssl_config.verifypeer)
+  if(conn->ssl_config.verifypeer &&
+     (DEFAULT_MANUAL_CRED_VALIDATION ||
+     (conn->ssl_config.CAfile && CAFILE_SUPPORTED))) {
```

Comment:
Thanks for this suggestion, I went ahead added the flag, `use_manual_cred_validation`,  to ssh_connect_data for Schannel to store the result of checking these conditions in schannel_connect_step1.

This check is now:
`if(conn->ssl_config.verifypeer && connssl->use_manual_cred_validation) {`

I figured this would be acceptable since a similar approach is used for the `use_alpn` flag.